### PR TITLE
Add more of the supported Processing algorithm parameter types to the writers guidelines

### LINKED ISSUE #3536: [FEATURE][processing] New parameter types for layouts and layout items
Original commit: https://github.com/qgis/QGIS/commit/021d1ef237065dd40c80b9b5044406f6319b18e0 by nyalldawson

Allows processing algorithms which operate on a specific layout
and layout item selection

### LINKED ISSUE #3540: [processing][FEATURE] New parameter type for map scales
Original commit: https://github.com/qgis/QGIS/commit/32fbf0a92b772a6ac7d9f4c41ff815d7dac80e90 by nyalldawson

This adds a new parameter type specifically for map scales,
QgsProcessingParameterScale. The values are evaluated using
self.parameterAsDouble, which returns the map scale
denominator (matching the standard in other parts of the
QGIS API).

Scale parameters are displayed to users using the standard
QgsScaleWidget, which includes the combo box of predefined
scales and a shortcut button to match the current map scale.

### LINKED ISSUE #3925: [FEATURE][processing] New parameter type for colors
Original commit: https://github.com/qgis/QGIS/commit/38bb5fb26087531d421895ddd80ee9375365c3fc by nyalldawson

Allows for algorithms which require a color value as an input

### PR DESCRIPTION
**Pull request by havatv**
Goal: Include the rest of the parameter types in the documentation writing guidelines.
Above the "line" I have added those that I found in
> python/core/auto_generated/processing/qgsprocessingparameters.sip.in

and

> src/core/processing/qgsprocessingparameters.h

Below the line are types added elsewhere (output from `find` on master).

> ./python/plugins/processing/algs/qgis/ui/InterpolationWidgets.py
> ./python/plugins/processing/algs/otb/OtbChoiceWidget.py:class OtbParameterChoice(QgsProcessingParameterDefinition):
> ./python/plugins/processing/algs/qgis/Aggregate.py:        class ParameterAggregates(QgsProcessingParameterDefinition):
> ./python/plugins/processing/algs/qgis/ExecuteSQL.py:class ParameterExecuteSql(QgsProcessingParameterDefinition):
> ./python/plugins/processing/algs/qgis/ui/InterpolationWidgets.py:class ParameterInterpolationData(QgsProcessingParameterDefinition):
> ./python/plugins/processing/algs/qgis/FieldsMapper.py:    class ParameterFieldsMapping(QgsProcessingParameterDefinition):
> ./python/plugins/processing/algs/qgis/Relief.py:class ParameterReliefColors(QgsProcessingParameterDefinition):

Ticket(s): #4426
Fixes #3540 (scale). Fixes #3925 (color). Fixes #3536 (layout).
Fixes: #3252 (authconfig)
#2579 (SQL)

- [ ] Backport to LTR documentation is required

### CODE REVIEW

**Comment by DelazJ** [comment]
I'm not sure I understand the purpose of this PR: you are adding every types in the code. Even if we would not need it? I'm particularly referring to the ones below the line: they are aggregates of sub types that we describe: eg, https://docs.qgis.org/testing/en/docs/user_manual/processing_algs/qgis/vectorgeometry.html#aggregate or https://docs.qgis.org/testing/en/docs/user_manual/processing_algs/qgis/rasterterrainanalysis.html#relief

**Comment by havatv** [comment]
@DelazJ, the purpose of this PR is to update the set of processing algrithm parameter types that we have "shortcuts" for in the documentation.
It would also be nice if we could have a discussion about (and perhaps decide) which parameter types it is worth to create "shortcuts" for - perhaps we can end up with some principles for deciding.
Thank you for contributing to the discussion.
Who else should have a say here?

**Comment by DelazJ** [comment]
What do you call "shortcut"?
> Who else should have a say here?

Everybody who is interested in the topic, I assume.

File: source/docs/documentation_guidelines/writing.rst [hunk 1]
```diff
@@ -622,7 +622,22 @@ guidelines:
...
+  Same output type as input type            ``same as input``
+  Definition                                ``definition``
+  Point                                     ``point``
+  MultipleLayers                            ``multipleLayers``
+  Range                                     ``Range``
+  AuthConfig                                ``AuthConfig``
+  Mesh                                      ``Mesh``
+  Layout                                    ``Layout``
+  LayoutItem                                ``LayoutItem``
+  Color                                     ``Color``
```
Comment (havatv) [inline source/docs/documentation_guidelines/writing.rst]:
```suggestion
  Color                                     ``Color``
  scale                                     ``scale``
```

Comment (havatv) [reply source/docs/documentation_guidelines/writing.rst]: Found a new one (#3540)

File: source/docs/documentation_guidelines/writing.rst [hunk 2]
```diff
@@ -622,7 +622,23 @@ guidelines:
...
+  scale                                     ``scale``
```
Comment (havatv) [inline source/docs/documentation_guidelines/writing.rst]:
```suggestion
  Scale                                     ``scale``
```

**Comment by havatv** [comment]
> What do you call "shortcut"?

Shortcut is the "code word" that we place in brackets ([]), for instance "[raster]", "[number]", "[crs]", and "[vector: any]".
When a user sees "[crs]" in the parameter table he/she will expect to get the standard widget for choosing a coordinate reference system when running the algorithm through the GUI (from the processing toolbox).  If we add the "[color]" shortcut, the user will expect to get a colour selection widget when seeing "[color]" in the parameter table.

> > Who else should have a say here?
> 
> Everybody who is interested in the topic, I assume.

Any of the developers that you think could be interested?

File: source/docs/documentation_guidelines/writing.rst [hunk 3]
```diff
@@ -622,7 +622,23 @@ guidelines:
   File                                      ``file``
   Matrix                                    ``matrix``
   Layer                                     ``layer``
-  Same output type as input type            ``same as input``  
+  Same output type as input type            ``same as input``
+  Definition                                ``definition``
+  Point                                     ``point``
+  MultipleLayers                            ``multipleLayers``
+  Range                                     ``Range``
```
Comment (havatv) [inline source/docs/documentation_guidelines/writing.rst]:
```suggestion
  Range                                     ``range``
```

File: source/docs/documentation_guidelines/writing.rst [hunk 4]
```diff
@@ -622,7 +622,23 @@ guidelines:
...
+  AuthConfig                                ``AuthConfig``
```
Comment (havatv) [inline source/docs/documentation_guidelines/writing.rst]:
```suggestion
  AuthConfig                                ``authconfig``
```

File: source/docs/documentation_guidelines/writing.rst [hunk 5]
```diff
@@ -622,7 +622,23 @@ guidelines:
...
+  Mesh                                      ``Mesh``
```
Comment (havatv) [inline source/docs/documentation_guidelines/writing.rst]:
```suggestion
  Mesh                                      ``mesh``
```

File: source/docs/documentation_guidelines/writing.rst [hunk 6]
```diff
@@ -622,7 +622,23 @@ guidelines:
...
+  Layout                                    ``Layout``
```
Comment (havatv) [inline source/docs/documentation_guidelines/writing.rst]:
```suggestion
  Layout                                    ``layout``
```

File: source/docs/documentation_guidelines/writing.rst [hunk 7]
```diff
@@ -622,7 +622,23 @@ guidelines:
...
+  LayoutItem                                ``LayoutItem``
```
Comment (havatv) [inline source/docs/documentation_guidelines/writing.rst]:
```suggestion
  LayoutItem                                ``layoutitem``
```

File: source/docs/documentation_guidelines/writing.rst [hunk 8]
```diff
@@ -622,7 +622,23 @@ guidelines:
...
+  Color                                     ``Color``
```
Comment (havatv) [inline source/docs/documentation_guidelines/writing.rst]:
```suggestion
  Color                                     ``color``
```

**Comment by havatv** [comment]
I think we should merge this. There can't be any harm in listing algorithm parameter types.

**Comment by havatv** [comment]
#6045 covers the parameters that were not added with #5044

**Pull request closed without merging**